runtime/compiler/runtime/JITServerAOTDeserializer.cpp

```diff
@@ -853,24 +877,28 @@ JITServerAOTDeserializer::updateSCCOffsets(SerializedAOTMethod *method, TR::Comp
       {
       // Get the SCC offset of the corresponding entity in the local SCC
       const SerializedSCCOffset &serializedOffset = method->offsets()[i];
-      uintptr_t sccOffset = getSCCOffset(serializedOffset.recordType(), serializedOffset.recordId(), wasReset);
-      if (sccOffset == (uintptr_t)-1)
-         return false;
-
-      // Update the SCC offset stored in AOT method relocation data
-      uint8_t *ptr = start + serializedOffset.reloDataOffset();
-      TR_ASSERT_FATAL((ptr >= start + sizeof(uintptr_t)/*skip the size word*/) && (ptr < end),
-                      "Out-of-bounds relocation data offset %zu in serialized method %s",
-                      serializedOffset.reloDataOffset(), comp->signature());
+      // Thunks do not use their SCC offset entries
+      if (serializedOffset.recordType() != Thunk)
+         {
+         uintptr_t sccOffset = getSCCOffset(serializedOffset.recordType(), serializedOffset.recordId(), wasReset);
+         if (sccOffset == (uintptr_t)-1)
```

Comment:
Under what conditions the sccOffset turns out to be -1?

Reply:
Tracking the code, I think that sccOffset can be set -1 if a concurrent reset is happening. A reset() is triggered when the client detects that it connected to a different server.
We should probably document somewhere all this so that we don't have to chase down the code.

Reply:
I think it can also become -1 when there's a ROMClass mismatch in `JITServerAOTDeserializer::cacheRecord(const ClassSerializationRecord *record, TR::Compilation *comp, bool &isNew, bool &wasReset);`